src/fu-remote-list.c

```diff
@@ -185,6 +186,47 @@ _fwupd_remote_build_component_id(FwupdRemote *remote)
 	return g_strdup_printf("org.freedesktop.fwupd.remotes.%s", fwupd_remote_get_id(remote));
 }
 
+static gboolean
+fu_remote_list_cleanup_remote(FwupdRemote *remote, GError **error)
+{
+	const gchar *fn_cache = fwupd_remote_get_filename_cache(remote);
+	g_autofree gchar *dirname = NULL;
+	g_autoptr(GPtrArray) files = NULL;
+
+	/* sanity check */
+	if (fn_cache == NULL)
+		return TRUE;
+	if (!g_str_has_suffix(fn_cache, ".xz"))
+		return TRUE;
+	if (fwupd_remote_get_kind(remote) != FWUPD_REMOTE_KIND_DOWNLOAD)
+		return TRUE;
+
```

Comment:
Shouldn't be here additional check if remote is enabled?
At the moment it is cleaning old files for remotes which are disabled in configuration files.

Reply:
Good catch fixed.